strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/main/java/org/carlspring/strongbox/authentication/external/ldap/LdapConfiguration.java

```diff
@@ -0,0 +1,216 @@
+        }
+        if (authenticator.getUserDnPatterns() != null)
+        {
+            bindAuthenticator.setUserDnPatterns(authenticator.getUserDnPatterns()
+                                                             .getUserDnPattern()
+                                                             .stream()
+                                                             .map(p -> p.getValue())
+                                                             .collect(Collectors.toList())
+                                                             .toArray(new String[0]));
+        }
+
+        DefaultLdapAuthoritiesPopulator defaultLdapAuthoritiesPopulator = new DefaultLdapAuthoritiesPopulator(contextSource,
+                                                                                                              authoritiesPopulator.getGroupSearchBase());
+
+        defaultLdapAuthoritiesPopulator.setGroupSearchFilter(authoritiesPopulator.getGroupSearchFilter());
+        defaultLdapAuthoritiesPopulator.setConvertToUpperCase(authoritiesPopulator.isConvertToUpperCase());
+        defaultLdapAuthoritiesPopulator.setGroupRoleAttribute(authoritiesPopulator.getGroupRoleAttribute());
+        defaultLdapAuthoritiesPopulator.setRolePrefix(authoritiesPopulator.getRolePrefix());
+        defaultLdapAuthoritiesPopulator.setSearchSubtree(authoritiesPopulator.isSearchSubtree());
+
+        AuthoritiesExternalToInternalMapper authoritiesMapper = new AuthoritiesExternalToInternalMapper();
+        authoritiesMapper.setRolesMapping(rolesMapping.asMap());
+        authoritiesMapper.afterPropertiesSet();
+        applicationContext.getParent().getAutowireCapableBeanFactory().autowireBean(authoritiesMapper);
+        registerSingleton(applicationContext, authoritiesMapper);
+
+        LdapAuthenticationProvider ldapAuthenticationProvider = new LdapAuthenticationProvider(bindAuthenticator,
+                                                                                               defaultLdapAuthoritiesPopulator);
+        ldapAuthenticationProvider.setAuthoritiesMapper(authoritiesMapper);
+
+
+        registerSingleton(applicationContext, ldapAuthenticationProvider);
+    }
+
+    private DefaultSpringSecurityContextSource registerLdapContainer(final GenericApplicationContext applicationContext)
+    {
+        String root = LdapUtils.parseRootDnFromUrl(url);
+        String port = getDefaultPort();
+
+        ApacheDSContainer container;
+        try
+        {
+            container = new ApacheDSContainer(root, ldif);
+            container.setPort(Integer.valueOf(port));
+            container.afterPropertiesSet();
+        }
+        catch (Exception ex)
+        {
+            throw Throwables.propagate(ex);
+        }
+        registerSingleton(applicationContext, container);
+
+        DefaultSpringSecurityContextSource contextSource = new DefaultSpringSecurityContextSource("ldap://127.0.0.1:" +
```

Comment:
Uhm... What is this about and why is it hardcoded?

Reply:
Yes, but this is how the Spring solution also works (as pointed above: `org.springframework.security.config.ldap.LdapServerBeanDefinitionParser#createEmbeddedServer`)